### PR TITLE
Added custom aliases for the hub command, adjustable in the config

### PR DESCRIPTION
**Pull request by thisisfel1x**
This pull request includes:

- [ ] breaking changes
- [X] no breaking changes

### Changes made to the repository:

Added an option for customizable hub command aliases in the config

~~Activate or deactivate the whole command with `"hubCommandEnabled": true/false`~~
Add or remove aliases by editing an array: `"hubCommandAliases": [
		"hub", "return", "l"
	],`
The command will be disabled when there are no aliases configured.

Implemented for BungeeCord and Velocity

Idk why but the `this` accessor was lost during the development

### CODE REVIEW

**Comment by byRoadrunner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263&bid=18925350)

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 1]
```diff
@@ -23,15 +23,20 @@
     private Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations;
     private Map<String, String> messages;
     private boolean logPlayerConnections = true;
+    private boolean hubCommandEnabled = true;
+    private ArrayList<String> hubCommandAliases;
 
     public BridgeConfiguration(String prefix, boolean onlyProxyProtection, Collection<String> excludedOnlyProxyWalkableGroups, Collection<String> excludedGroups,
-                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages, boolean logPlayerConnections) {
+                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages,
+                               boolean logPlayerConnections, boolean hubCommandEnabled, ArrayList<String> hubCommandAliases) {
```
Comment (byRoadrunner) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:31]: Codacy found an issue: [Avoid unused constructor parameters such as 'logPlayerConnections'.](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263)

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 2]
```diff
@@ -23,15 +23,20 @@
     private Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations;
     private Map<String, String> messages;
     private boolean logPlayerConnections = true;
+    private boolean hubCommandEnabled = true;
+    private ArrayList<String> hubCommandAliases;
```
Comment (derrop) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:27]: Please use the Collection interface instead of an ArrayList here

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java (hunk 1 above)
Comment (derrop) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:31]: The same as above, Collection instead of ArrayList.

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java [hunk 3]
```diff
@@ -15,14 +15,12 @@
...
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 
 public final class CloudNetBridgeModule extends NodeCloudNetModule {
 
     private static final Map<String, String> DEFAULT_MESSAGES = new HashMap<>();
+    private static final ArrayList<String> DEFAULT_HUB_COMMAND_ALIASES = new ArrayList<>();
```
Comment (derrop) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java:23]: Same as above, use Collection instead of an ArrayList.
And you can also use Arrays.asList instead of calling Collection#add multiple times.

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java [hunk 4]
```diff
@@ -72,8 +77,18 @@ public void createConfiguration() {
...
         }
 
-        this.getConfig().append("config", this.bridgeConfiguration);
-        this.saveConfig();
+        if (this.bridgeConfiguration.getLobbyCommandAliases() != null) {
+            for (String lobbyCommandAlias : this.bridgeConfiguration.getLobbyCommandAliases()) {
+                if(!this.bridgeConfiguration.getLobbyCommandAliases().contains(lobbyCommandAlias)) {
+                    this.bridgeConfiguration.getLobbyCommandAliases().add(lobbyCommandAlias);
+                }
+            }
```
Comment (derrop) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java:85]: This whole loop seems unnecessary to me, why does it exist?

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java (hunk 2 above)
Comment (juliarn) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:27]: List*

Comment (derrop) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:27]: Why not a Collection?

Comment (juliarn) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:27]: Because the order is important (to get the name the command should have)

Comment (derrop) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:27]: The name doesn't matter, and Velocity doesn't even differentiate between the name and aliases.

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetBridgePlugin.java [hunk 5]
```diff
@@ -73,13 +73,16 @@ private void initListeners() {
 
         //CloudNet
         CloudNetDriver.getInstance().getEventManager().registerListener(new VelocityCloudNetListener());
-        CloudNetDriver.getInstance().getEventManager().registerListener(new VelocityPlayerExecutorListener(this.proxyServer));
```
Comment (derrop) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetBridgePlugin.java:76]: Why did you remove this?

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/BungeeCloudNetBridgePlugin.java [hunk 6]
```diff
@@ -86,7 +91,6 @@ private void initListeners() {
 
         //CloudNet
         CloudNetDriver.getInstance().getEventManager().registerListener(new BungeeCloudNetListener());
-        CloudNetDriver.getInstance().getEventManager().registerListener(new BungeePlayerExecutorListener());
```
Comment (derrop) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/BungeeCloudNetBridgePlugin.java:89]: Same as for the VelocityPlayerExecutorListener

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetBridgePlugin.java (hunk 5 above)
Comment (thisisfel1x) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetBridgePlugin.java:76]: I didn't even noticed
Well, I started editing the master branch, then switch to development. Now it should be clear why some stuff is missing, like the `this` accessor.
Sorry, I'll fix that :/

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/command/CommandHub.java [hunk 7]
```diff
@@ -42,6 +42,6 @@ public void execute(CommandSender sender, String[] args) {
 
     @Override
     public String[] getAliases() {
-        return new String[]{"lobby", "l", "leave"};
+        return BridgeConfigurationProvider.load().getLobbyCommandAliases().toArray(new String[0]);
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/command/CommandHub.java:45]: Maybe save this as a array inside of the class instead of re-creating it every time

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java [hunk 8]
```diff
@@ -72,8 +77,18 @@ public void createConfiguration() {
             this.bridgeConfiguration.setMessages(new HashMap<>(DEFAULT_MESSAGES));
         }
 
-        this.getConfig().append("config", this.bridgeConfiguration);
-        this.saveConfig();
+        if (this.bridgeConfiguration.getLobbyCommandAliases() != null) {
+            for (String lobbyCommandAlias : this.bridgeConfiguration.getLobbyCommandAliases()) {
+                if(!this.bridgeConfiguration.getLobbyCommandAliases().contains(lobbyCommandAlias)) {
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java:82]: if (

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java [hunk 9]
```diff
@@ -85,40 +100,40 @@ public ProxyFallbackConfiguration createDefaultFallbackConfiguration(String targ
     }
 
     public void writeConfiguration(BridgeConfiguration bridgeConfiguration) {
-        this.getConfig().append("config", bridgeConfiguration);
-        this.saveConfig();
+        getConfig().append("config", bridgeConfiguration);
+        saveConfig();
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java:104]: Why do you remove the this accessors?

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetBridgePlugin.java [hunk 10]
```diff
@@ -73,13 +73,16 @@ private void initListeners() {
...
         CloudNetDriver.getInstance().getEventManager().registerListener(new BridgeCustomChannelMessageListener());
     }
 
     private void registerCommands() {
         this.proxyServer.getCommandManager().register(new CommandCloudNet(), "cloudnet", "cloud", "cl");
-        this.proxyServer.getCommandManager().register(new CommandHub(), "hub", "l", "leave", "lobby");
+        if(BridgeConfigurationProvider.load().isHubCommandEnabled()
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetBridgePlugin.java:81]: if (

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetBridgePlugin.java [hunk 11]
```diff
@@ -91,7 +94,7 @@ private void initServers() {
                 }
 
                 String name = serviceInfoSnapshot.getServiceId().getName();
-                this.proxyServer.registerServer(new ServerInfo(name, new InetSocketAddress(
+                proxyServer.registerServer(new ServerInfo(name, new InetSocketAddress(
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetBridgePlugin.java:97]: Same here use `this`

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 12]
```diff
@@ -93,4 +98,15 @@ public void setLogPlayerConnections(boolean logPlayerConnections) {
         this.logPlayerConnections = logPlayerConnections;
     }
 
+    public ArrayList<String> getLobbyCommandAliases() {
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:101]: @NotNull Collection<String>

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 13]
```diff
@@ -23,15 +23,20 @@
...
-                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages, boolean logPlayerConnections) {
+                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages,
+                               boolean logPlayerConnections, boolean hubCommandEnabled, ArrayList<String> hubCommandAliases) {
         this.prefix = prefix;
         this.onlyProxyProtection = onlyProxyProtection;
         this.excludedOnlyProxyWalkableGroups = excludedOnlyProxyWalkableGroups;
         this.excludedGroups = excludedGroups;
         this.bungeeFallbackConfigurations = bungeeFallbackConfigurations;
         this.messages = messages;
+        this.hubCommandEnabled = hubCommandEnabled;
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:38]: Maybe instead of an extra boolen if the aliases are empty? @derrop @juliarn ?

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 14]
```diff
@@ -46,7 +51,7 @@ public void setPrefix(String prefix) {
     }
 
     public boolean isOnlyProxyProtection() {
-        return this.onlyProxyProtection;
+        return onlyProxyProtection;
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:54]: Do not remove `this`

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/BungeeCloudNetBridgePlugin.java [hunk 15]
```diff
@@ -77,7 +77,12 @@ private void initServers() {
 
     private void registerCommands() {
         ProxyServer.getInstance().getPluginManager().registerCommand(this, new CommandCloudNet());
-        ProxyServer.getInstance().getPluginManager().registerCommand(this, new CommandHub());
+
+        if(BridgeConfigurationProvider.load().isHubCommandEnabled()
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/BungeeCloudNetBridgePlugin.java:81]: if (

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/command/CommandHub.java [hunk 16]
```diff
@@ -11,7 +11,7 @@
 public final class CommandHub extends Command {
 
     public CommandHub() {
-        super("hub");
+        super(BridgeConfigurationProvider.load().getLobbyCommandAliases().stream().findFirst().orElse("hub"));
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/command/CommandHub.java:14]: Just give here the array if it is not-empty and use the first array entry

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java [hunk 17]
```diff
@@ -32,6 +30,11 @@
         DEFAULT_MESSAGES.put("server-join-cancel-because-maintenance", "&7This server is currently in maintenance mode");
         DEFAULT_MESSAGES.put("command-cloud-sub-command-no-permission", "&7You are not allowed to use &b%command%");
         DEFAULT_MESSAGES.put("already-connected", "§cYou are already connected to this network!");
+
+        DEFAULT_HUB_COMMAND_ALIASES.add("lobby");
+        DEFAULT_HUB_COMMAND_ALIASES.add("hub");
+        DEFAULT_HUB_COMMAND_ALIASES.add("leave");
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java:36]: addAll...

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java [hunk 18]
```diff
@@ -52,14 +55,16 @@ public static CloudNetBridgeModule getInstance() {
     public void createConfiguration() {
         this.getModuleWrapper().getDataFolder().mkdirs();
 
-        this.bridgeConfiguration = this.getConfig().get("config", BridgeConfiguration.TYPE, new BridgeConfiguration(
+        this.bridgeConfiguration = getConfig().get("config", BridgeConfiguration.TYPE, new BridgeConfiguration(
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java:58]: Do not remove `this`

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java [hunk 19]
```diff
@@ -52,14 +55,16 @@ public static CloudNetBridgeModule getInstance() {
...
                 "&7Cloud &8| &b",
                 true,
                 new ArrayList<>(),
                 new ArrayList<>(),
                 new ArrayList<>(),
                 DEFAULT_MESSAGES,
-                true
+                true,
+                true,
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java:66]: See down if this boolean is really needed

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java (hunk 8 above)
Comment (juliarn) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java:82]: Yeah formatting should be done

**Comment by byRoadrunner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263&bid=18925350)

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 20]
```diff
@@ -23,15 +23,20 @@
     private Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations;
     private Map<String, String> messages;
     private boolean logPlayerConnections = true;
+    private boolean hubCommandEnabled = true;
+    private Collection<String> hubCommandAliases;
 
     public BridgeConfiguration(String prefix, boolean onlyProxyProtection, Collection<String> excludedOnlyProxyWalkableGroups, Collection<String> excludedGroups,
-                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages, boolean logPlayerConnections) {
+                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages,
+                               boolean logPlayerConnections, boolean hubCommandEnabled, Collection<String> hubCommandAliases) {
```
Comment (byRoadrunner) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:31]: Codacy found an issue: [Avoid unused constructor parameters such as 'logPlayerConnections'.](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263)

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java (hunk 9 above)
Comment (thisisfel1x) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java]: I've already wrote a message about this problem. I started editing the master branch 😬 and then moved to the developement branch. I shouldn't use copy and paste xD

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 21]
```diff
@@ -24,11 +24,11 @@
     private Map<String, String> messages;
     private boolean logPlayerConnections = true;
     private boolean hubCommandEnabled = true;
-    private ArrayList<String> hubCommandAliases;
+    private Collection<String> hubCommandAliases;
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:27]: This may be final

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 22]
```diff
@@ -98,14 +98,10 @@ public void setLogPlayerConnections(boolean logPlayerConnections) {
         this.logPlayerConnections = logPlayerConnections;
     }
 
-    public ArrayList<String> getLobbyCommandAliases() {
+    public Collection<String> getLobbyCommandAliases() {
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:101]: @NotNull ...

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java (hunk 13 above)
Comment (thisisfel1x) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:38]: Personally I like the option with an extra boolean more. Maybe this guarantees fewer problems with the user than telling him that he has to remove everything in the array for disabling the command.

Comment (derklaro) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:38]: As you can see rop and julian have the same opinion, I would also prefer that to bool

**Comment by byRoadrunner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263&bid=18925350)

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 23]
```diff
@@ -19,22 +20,24 @@
...
     private Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations;
     private Map<String, String> messages;
     private boolean logPlayerConnections = true;
+    private boolean hubCommandEnabled = true;
+    private final Collection<String> hubCommandAliases;
 
     public BridgeConfiguration(String prefix, boolean onlyProxyProtection, Collection<String> excludedOnlyProxyWalkableGroups, Collection<String> excludedGroups,
-                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages, boolean logPlayerConnections) {
+                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages,
+                               boolean logPlayerConnections, boolean hubCommandEnabled, Collection<String> hubCommandAliases) {
```
Comment (byRoadrunner) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:32]: Codacy found an issue: [Avoid unused constructor parameters such as 'logPlayerConnections'.](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263)

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 24]
```diff
@@ -19,22 +20,24 @@
 
     private String prefix;
     private boolean onlyProxyProtection = true;
-    private Collection<String> excludedOnlyProxyWalkableGroups = new ArrayList<>(), excludedGroups;
+    private Collection<String> excludedOnlyProxyWalkableGroups, excludedGroups;
```
Comment (byRoadrunner) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:23]: Codacy found an issue: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263)

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/command/CommandHub.java [hunk 25]
```diff
@@ -10,8 +10,10 @@
 
 public final class CommandHub extends Command {
 
+    String[] aliases = BridgeConfigurationProvider.load().getLobbyCommandAliases().toArray(new String[0]);
```
Comment (byRoadrunner) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/bungee/command/CommandHub.java:13]: Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263)

**Comment by thisisfel1x** [comment]
I'm currently not convinced to remove the extra boolean to enable/disable the command. This makes it easier to use and you wouldn't have to ask the support how to disable the command

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java (hunk 21 above)
Comment (derrop) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java]: Anything else in this class isn't final either, I think a setter would be better. Without a setter, it would just be more difficult to modify the aliases.

**Comment by byRoadrunner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           

Complexity decreasing per file
==============================
+ cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263&bid=18925350)

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java [hunk 26]
```diff
@@ -16,14 +16,12 @@
...
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 
 public final class CloudNetBridgeModule extends NodeCloudNetModule {
 
     private static final Map<String, String> DEFAULT_MESSAGES = new HashMap<>();
+    private static final Collection<String> DEFAULT_HUB_COMMAND_ALIASES = Arrays.asList("hub", "lobby", "leave", "l");
```
Comment (byRoadrunner) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java:24]: Codacy found an issue: [Avoid unused private fields such as 'DEFAULT_HUB_COMMAND_ALIASES'.](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263)

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java [hunk 27]
```diff
@@ -6,26 +6,46 @@
...
+    private List<String> hubCommandNames = Arrays.asList("hub", "lobby", "leave", "l");
+
     private boolean logPlayerConnections = true;
 
+    private Map<String, String> messages = DEFAULT_MESSAGES;
+
     public BridgeConfiguration(String prefix, boolean onlyProxyProtection, Collection<String> excludedOnlyProxyWalkableGroups, Collection<String> excludedGroups,
-                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages, boolean logPlayerConnections) {
+                               Collection<ProxyFallbackConfiguration> bungeeFallbackConfigurations, Map<String, String> messages,
+                               boolean logPlayerConnections) {
```
Comment (byRoadrunner) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/BridgeConfiguration.java:48]: Codacy found an issue: [Avoid unused constructor parameters such as 'logPlayerConnections'.](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263)

**Review by derrop — APPROVED** [review]

**Comment by byRoadrunner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity decreasing per file
==============================
+ cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/node/CloudNetBridgeModule.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5780263&bid=18925350)
